network/common/src/main/java/org/apache/spark/network/util/ByteUnit.java

```diff
@@ -0,0 +1,165 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.spark.network.util;
+
+/**
+ * Code based on https://github.com/fabian-barney/Utils
```

Comment:
We have to be a little careful if importing third-party code, especially source. Looks like this is AL2 licensed: https://github.com/fabian-barney/Utils/blob/master/utils/LICENSE  And it has no `NOTICE` file. So should be OK.

However, you can't remove his copyright statement. If you also added changes or new code, you can also include the ASF header to cover your work, but should retain his too.

However, I'm not sure this is the best code to reuse, and I base that almost entirely on the fact that it treats numbers of bytes as `double`s which seems wrong. This feels like something that doesn't need to be this complex, so could be reimplemented easily.